Build is currently broken (unresolved import `zstd_sys::ZSTD_cParameter::ZSTD_c_experimentalParam6`)

Building Hypermine from scratch results in the following error:

```
error[E0432]: unresolved import `zstd_sys::ZSTD_cParameter::ZSTD_c_experimentalParam6`
   --> C:\Users\Patrick\.cargo\registry\src\index.crates.io-6f17d22bba15001f\zstd-safe-6.0.6\src\lib.rs:609:13
    |
609 |             ZSTD_c_experimentalParam6 as ZSTD_c_targetCBlockSize,
    |             -------------------------^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             no `ZSTD_c_experimentalParam6` in `ZSTD_cParameter`
    |             help: a similar name exists in the module: `ZSTD_c_experimentalParam1`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `zstd-safe` (lib) due to 1 previous error
```

See https://github.com/gyscos/zstd-rs/issues/270 and https://github.com/gyscos/zstd-rs/issues/268 for the likely root cause.

This comes from the dependency we import in save/Cargo.toml:
```toml
zstd = { package = "zstd-safe", version = "6.0.4", features = ["std", "experimental"] }
```

### Current workaround

Add the following after the `zstd` dependency in save/cargo.toml:
```toml
zstd-sys = "=2.0.9"
```

--- a/common/src/chunk_collision.rs
+++ b/common/src/chunk_collision.rs
@@ -28,11 +28,8 @@ pub fn chunk_sphere_cast(
 ) -> Option<ChunkCastHit> {
     let mut hit: Option<ChunkCastHit> = None;
 
-    let Some(bounding_box) =
-        VoxelAABB::from_ray_segment_and_radius(layout, ray, tanh_distance, collider_radius)
-    else {
-        return None;
-    };
+    let bounding_box =
+        VoxelAABB::from_ray_segment_and_radius(layout, ray, tanh_distance, collider_radius)?;
 
     for t_axis in 0..3 {
         hit = find_face_collision(

--- a/common/src/chunk_ray_casting.rs
+++ b/common/src/chunk_ray_casting.rs
@@ -32,11 +32,7 @@ pub fn chunk_ray_cast(
 ) -> Option<ChunkCastHit> {
     let mut hit: Option<ChunkCastHit> = None;
 
-    let Some(bounding_box) =
-        VoxelAABB::from_ray_segment_and_radius(layout, ray, tanh_distance, 0.0)
-    else {
-        return None;
-    };
+    let bounding_box = VoxelAABB::from_ray_segment_and_radius(layout, ray, tanh_distance, 0.0)?;
 
     for t_axis in 0..3 {
         hit = find_face_collision(

--- a/save/Cargo.toml
+++ b/save/Cargo.toml
@@ -9,7 +9,7 @@ edition = "2021"
 prost = "0.12.2"
 redb = "1.0"
 thiserror = "1.0.38"
-zstd = { package = "zstd-safe", version = "6.0.4", features = ["std", "experimental"] }
+zstd = { package = "zstd-safe", version = "7.1.0", default-features = false, features = ["std", "experimental"] }
 
 [dev-dependencies]
 tempfile = "3.4"

--- a/save/src/lib.rs
+++ b/save/src/lib.rs
@@ -158,12 +158,12 @@ fn decompress(
 ) -> Result<(), &'static str> {
     dctx.init().map_err(zstd::get_error_name)?;
     let mut input = zstd::InBuffer::around(compressed);
-    let mut out = zstd::OutBuffer::around(out);
     let out_size = zstd::DCtx::out_size();
     loop {
-        if out.dst.len() + out_size > out.dst.capacity() {
-            out.dst.reserve(out_size);
+        if out.len() + out_size > out.capacity() {
+            out.reserve(out_size);
         }
+        let mut out = zstd::OutBuffer::around_pos(out, out.len());
         let n = dctx
             .decompress_stream(&mut out, &mut input)
             .map_err(zstd::get_error_name)?;

